src/openforms/formio/rendering/registry.py

```diff
@@ -1,19 +1,23 @@
-from typing import Type
+from typing import Callable, TypeAlias
 
 from openforms.plugins.registry import BaseRegistry
 
 from .nodes import ComponentNode
 
+TComponentNode: TypeAlias = type[ComponentNode]
```

Comment:
Maybe we should be more generic: `class Registry(BaseRegistry[TComponentNode])`?

And the binding to AbstractBasePlugin and iter_enabled_plugins, should live on some BasePluginRegistry? :thinking: 

Also the docstring of `__call__` doesn't make sense to me: all Registry's I've seen are used as class decorators.
The only differences I see in the override is the unique_identifier length isn't checked, and check_plugin isn't called, no reference to the registry is set on the registrant, and a runtime check on top of the type check.

Reply:
PS plugins.BaseRegistry is a fine name for such a "BasePluginRegistry"; no need to break api.

Reply:
quickly discussed on slack - `ComponentNode` should just conform to `AbstractBasePlugin`, I don't see a reason why it shouldn't, but I deliberately left it out of scope for this PR because it was getting too big.

There's some other formio rendering refactoring that needs to be done, so perhaps I can group all those things together (there's a draft ticket in my column on the project board)